io-sim-classes/src/Control/Monad/Tracer.hs

```diff
@@ -0,0 +1,32 @@
+{-# LANGUAGE OverloadedStrings      #-}
+
+module Control.Monad.Tracer
+  ( Tracer
+  , tracingWith
+  -- tracers
+  , sayTracer
+  , toDynListTracer
+  , toListTracer
+  , nullTracer
+  ) where
+
+import           Data.Functor.Contravariant (Op (..))
+import           Data.Typeable (Typeable)
+import           Data.Dynamic (Dynamic, toDyn)
+
+import           Cardano.BM.Tracer (Tracer (..), nullTracer,
+                     tracingWith)
+
+import           Control.Monad.Class.MonadSay
+import           Control.Monad.Class.MonadSTM
+
+sayTracer :: (Show a, MonadSay m) => Tracer m a
+sayTracer = Tracer $ Op $ say . show
+
+toListTracer :: (MonadSTM m) => TVar m [a] -> Tracer m a
```

Comment:
Maybe just `listTracer` and `dynListTracer`, when it will be used at call sites, it will look better I think.